src/main/java/org/osc/manager/ism/api/IsmDeviceApi.java

```diff
@@ -242,23 +252,39 @@ public String updateDeviceMember(ManagerDeviceMemberElement deviceElement, Strin
         DeviceMemberEntity deviceMember = new DeviceMemberEntity();
         deviceMember.setId(Long.parseLong(deviceElement.getId()));
         deviceMember.setName(name);
+        updateMember(deviceMember, vserverIpAddress, contactIpAddress, ipAddress, gateway, prefixLength);
         return updateDeviceMember(deviceMember, null, this.vs.getId());
-        // TODO - Sudhir Add other information like gateway, ipAddress
+    }
+
+    public void copyDeviceMember(DeviceMemberEntity destination, DeviceMemberEntity source) {
```

Comment:
use copy constructor

Reply:
I thought of using it initially and I remember seeing exception earlier (some days ago) from Hibernate that constructors should not part of entity element and hibernate throws exception will persisting the entity.
Will implement this and check and come back if I face any issues.

Reply:
put this code into the device member entity

Reply:
we cannot have a copy constructor in this case, as the scenario is as follows:
we have object 'A' which is populated by DB_READ and another object 'B' which is passed as part of method parameters. what we do is that object B is updated over object 'A' meaning some fields of object 'B' is updated over object 'A' and eventually object 'A' is written back to DB.

Reply:
will add a new update method in devicememberentity